Added Profile Picture, Edit Account Info, and Delete Account

MatthewLoyed opened the pull request:
I added the functionality to add your own profile picture, edit your account info, and delete your account. Improved and fixed bugs regarding registering account and accessing database info. Added dark mode toggle but needs work.

MatthewLoyed committed:
Added functionality to add profile picture, edit account info, and delete account. Improved and fixed bugs regarding registering account and accessing database info. Added dark mode toggle but needs work.

MatthewLoyed committed:
Added password requirements to Update Password.

MatthewLoyed committed:
Merge branch 'main' into feature/edit-account

MatthewLoyed committed:
Fix parsing errors and React imports in Account and CreateAccount pages

aambrose1 commented on client/src/pages/Account/Account.js:
Line 174 useEffect section, this is supposed to check the user database. Unless its easier just to get everyone from the treemember db?
Line 184, url should be api/family-members/member (search for member). Also double check the properties, I think they are all lowercase pulled directly from the db

aambrose1 replied:
I made the fetch changes on bug/edit-acc . Otherwise looks good

aambrose1 commented on client/src/pages/Account/Account.js:
UI looks good as well! It should change "My profile" when the user clicks on another account. You can test this directly by changing the params in the URL on the account page.

aambrose1 commented:
Also, merge in the main branch, the backend is out of date

MatthewLoyed committed:
Fix Account.js fetch URL and profile header

MatthewLoyed committed:
Merge branch 'main' into feature/edit-account

aambrose1 reviewed (changes requested):
View the data fetch alterations on bug/edit-acc to fix other supabase users and manual family member's info not showing up. And if those fixes work with you, we can merge them in or implement something like that.

MatthewLoyed committed:
fix(account): fetch auth users before falling back to family member endpoint

This ports the logic from the bug/edit-acc branch to correctly load profile information for registered supabase users when viewing their profiles.

aambrose1 reviewed (approved):
Looks good. Merging into main
